src/extractor/index.ts

```diff
@@ -12,10 +12,22 @@ export type ExtractedKey = Key & {
 
 export type Warning = { warning: string; line: number };
 
+export type VerboseOption = 'extractor';
+
+export type ExtractOptions = {
+  verbose?: VerboseOption[];
+  strictNamespace: boolean;
+  defaultNamespace: string | undefined;
+};
+
+export type ParserType = 'react' | 'vue' | 'svelte';
+
 export type Extractor = (
   fileContents: string,
-  fileName: string
-) => ExtractionResult[];
+  fileName: string,
+  parserType: ParserType,
+  options: ExtractOptions
+) => ExtractionResult;
```

Comment:
This type was originally meant to represent any extractor including custom extractors; I'm not sure extractors that aren't the internal one would receive the parser type